File: LC100/lc69.js
function sqrt(x) {
    if (x === 0) {
        return 0;
    }
    var low = 0, high = x, mid;
    while (low <= high) {
        mid = low + Math.floor((high - low)/2);
        if (mid * mid > x) {
            high = mid - 1;
        } else {
            if ((mid + 1) * (mid + 1) > x) {
                return mid;
            }
            low = mid + 1;
        }
    }
}


console.log(sqrt(0));
console.log(sqrt(1));
console.log(sqrt(3));
console.log(sqrt(10));
console.log(sqrt(48));

File: LC100/lc69.py
"""
Implement int sqrt(int x).

Compute and return the square root of x.
"""


def sqrt(number):
    if number == 0:
        return 0
    low = 0
    high = number
    while low <= high:
        mid = low + (high - low) // 2
        if mid * mid > number:
            high = mid - 1
        else:
            if (mid + 1) * (mid + 1) > number:
                return mid
            low = mid + 1


print(sqrt(0))
print(sqrt(1))
print(sqrt(3))
print(sqrt(10))
print(sqrt(48))
